internal/ignition/ignition.go

```diff
@@ -885,13 +887,18 @@ func (g *installerGenerator) updateBootstrap(ctx context.Context, bootstrapPath
 				// drop this from the list of Files because we don't want to run BMO
 				continue
 			}
+			if provErr := removeProvisioningPausedAnnotation(&config.Storage.Files[i]); err != nil {
```

Comment:
Currently, whenever assisted is creating a BMH resource the BMH status is `unmanaged` (SaaS) or `externally provisioned` (ZTP). if assisted behavior change or ironic/metal3 will try to update the BMh status we will end up with the original issue (https://issues.redhat.com//browse/OCPBUGS-33157).
Perhaps removing this annotation from the assited-installer-controller after all control plane nodes are ready is preferable since it aligns with the desired CBO behavior.

Reply:
The annotation is purely to prevent BMO in the cluster interfering with the BMO on the baremetal bootstrap host while it is still doing its work.
In a system with no baremetal bootstrap host it is not required.